go/tunnels/client.go

```diff
@@ -136,33 +136,26 @@ func (c *Client) Connect(ctx context.Context, hostID string) error {
 	return nil
 }
 
-// Opens a stream connected to a remote port for clients which cannot or do not want to forward local TCP ports.
-// Returns a readWriteCloser which can be used to read and write to the remote port.
-// Set AcceptLocalConnectionsForForwardedPorts to false in ConnectAsync to ensure TCP listeners are not created
-// This will return an error if the port is not yet forwarded,
-// the caller should first call WaitForForwardedPort.
-func (c *Client) ConnectToForwardedPort(ctx context.Context, listenerIn *net.TCPListener, port uint16) error {
+// ConnectListenerToForwardedPort opens a stream to a remote port and connects it to a given listener.
+//
+// Ensure that the port is already forwarded before calling this function
+// by calling WaitForForwardedPort. Otherwise, this will return an error.
+//
+// Set acceptLocalConnectionsForForwardedPorts to false when creating the client to ensure
+// TCP listeners are not created for all ports automatically when the client connects.
+func (c *Client) ConnectListenerToForwardedPort(ctx context.Context, listenerIn net.Listener, port uint16) error {
```

Comment:
Updated the `listenerIn` type to be the generic `net.Listener` in case users don't want to use TCP.